### PR TITLE
docs: add scrollBehavior API detail and link to guide

### LINKED ISSUE #953: scrollBehavior API docs would benefit from a link to the advanced guide
### What problem does this feature solve?
I typically go straight to the API docs for Vue tools, but the `scrollBehavior` documentation in vue-router-next is pretty minimal. I didn't realise until I'd spent some time looking at the code that there's a very comprehensive guide available in the docs.

I'm happy to contribute a PR for this very small change.

### scrollBehavior

Function to control scrolling when navigating between pages. Can return:
  - a ScrollPosition object, which can contain:
    - [ScrollToOptions](https://developer.mozilla.org/en-US/docs/Web/API/ScrollToOptions)
    - an `el` property with a selector or reference to an element
  - a Promise to delay scrolling
  
See [Scroll Behaviour](../guide/advanced/scroll-behavior.md) for more details.

**Signature:**

```typescript
scrollBehavior?: RouterScrollBehavior
```

#### Examples

```js
function scrollBehavior(to, from, savedPosition) {
  // `to` and `from` are both route locations
  // `savedPosition` can be null if there isn't one
  return savedPosition || { top: 0 };
}
```

### PR DESCRIPTION
**Pull request by aethr**
Fixes #953 

Add a bit more detail to the `scrollBehavior` API docs, and most importantly link to the very comprehensive guide.

### CODE REVIEW

File: docs/api/index.md [hunk 1]
```diff
@@ -760,6 +766,7 @@ scrollBehavior?: ScrollBehavior
 function scrollBehavior(to, from, savedPosition) {
   // `to` and `from` are both route locations
   // `savedPosition` can be null if there isn't one
+  return savedPosition || { top: 0 };
```
Comment (posva) [inline docs/api/index.md:769]: this is just the API, let's remove this

```suggestion
```

File: docs/api/index.md [hunk 2]
```diff
@@ -746,12 +746,18 @@ routes: RouteRecordRaw[]
 
 ### scrollBehavior
 
-Function to control scrolling when navigating between pages. Can return a Promise to delay scrolling. Check .
+Function to control scrolling when navigating between pages. Can return:
+  - a ScrollPosition object, which can contain:
+    - [ScrollToOptions](https://developer.mozilla.org/en-US/docs/Web/API/ScrollToOptions)
+    - an `el` property with a selector or reference to an element
+  - a Promise to delay scrolling
```
Comment (posva) [inline docs/api/index.md:753]: Can you revert this change? This is already included in the guide

**Review by posva — CHANGES_REQUESTED** [review]

File: docs/api/index.md (hunk 1 above)
Comment (aethr) [reply docs/api/index.md:769]: Fair enough, since the heading was "example" I thought it might be helpful to show a return value. Removing.

File: docs/api/index.md (hunk 2 above)
Comment (aethr) [reply docs/api/index.md]: Just added a small bit of extra detail and removed the incomplete sentence.

File: docs/api/index.md [hunk 3]
```diff
@@ -746,12 +746,14 @@ routes: RouteRecordRaw[]
 
 ### scrollBehavior
 
-Function to control scrolling when navigating between pages. Can return a Promise to delay scrolling. Check .
+Function to control scrolling when navigating between pages. Can return a ScrollPosition object or a Promise to delay scrolling.
+  
+See [Scroll Behaviour](../guide/advanced/scroll-behavior.md) for more details.
```
Comment (posva) [inline docs/api/index.md:751]:
```suggestion
Function to control scrolling when navigating between pages. Can return a Promise to delay when the scrolling happens. See [Scroll Behaviour](../guide/advanced/scroll-behavior.md) for more details.
```

**Review by posva — APPROVED** [review]
Thanks!
